Solution task number BJS2-27206

ALEH-BIAHUN opened the pull request.

ALEH-BIAHUN committed:
Solution task number BJS2-27206

mikhailov26 commented on src/main/java/faang/school/projectservice/controller/stage/StageInvitationController.java:
```diff
@@ -0,0 +1,47 @@
+@RequestMapping("/api/invitations")
+public class StageInvitationController {
+
+    @Autowired
+    private StageInvitationService invitationService;
```
Прочитай почему через поле инжект не очень хорошо делать

mikhailov26 commented on src/main/java/faang/school/projectservice/service/StageInvitationService.java:
```diff
@@ -0,0 +1,108 @@
+@Service
+public class StageInvitationService {
+
+    @Autowired
+    private StageInvitationRepository invitationRepository;
+
+    @Autowired
+    private StageInvitationMapper mapper;
+
+    @Autowired
+    private StageRepository stageRepository;
+
+    @Autowired
+    private TeamMemberRepository teamMemberRepository;
+
```
Аналогично прошлому комменту

mikhailov26 commented on src/main/java/faang/school/projectservice/service/StageInvitationService.java:
```diff
@@ -0,0 +1,108 @@
+
+    public List<StageInvitationDto> getInvitationsByUser(Long userId) {
+        List<StageInvitation> allInvitations = invitationRepository.findAll();
+
+        List<StageInvitation> filteredInvitations = allInvitations.stream()
+                .filter(invitation -> invitation.getInvited().getId().equals(userId))
+                .collect(Collectors.toList());
+
+        return filteredInvitations.stream()
+                .map(mapper :: toDto)
+                .collect(Collectors.toList());
```
Подумай можно ли за один вызов .stream получить лист дто

ALEH-BIAHUN committed:
Solution task number BJS2-27206 v.2.0
Fixed the errors related to dependency injection in the StageInvitationService class and also reworked the getInvitationsByUser method.
